### PR TITLE
fix(rpc): Supports RPC calls for new identifier CustomResource

### PR DESCRIPTION
**Pull request by ankk13**
this identifier is introduced in BlobInfo and Result structs

### CODE REVIEW

File: rpc/common/service.proto [hunk 1]
```diff
@@ -128,3 +128,10 @@ message CVSS {
   double v2_score  = 3;
   double v3_score  = 4;
 }
+
+message CustomResource {
+  string                 type     = 1;
+  string                 filepath = 2;
+  Layer                  layer    = 3;
+  google.protobuf.Value  info     = 4;
```
Comment (knqyf263) [inline rpc/common/service.proto:136]: `data`, `value`, `content` or something like that may be better.

File: rpc/cache/service.proto [hunk 2]
```diff
@@ -58,3 +59,11 @@ message MissingBlobsResponse {
...
   repeated string missing_blob_ids = 2;
 }
+
+message GetBlobRequest {
+  string blob_id = 1;
+}
+
+message GetBlobResponse {
+  BlobInfo blob_info = 1;
+}
```
Comment (knqyf263) [inline rpc/cache/service.proto:69]: What are they for?

**Review by knqyf263 — COMMENTED** [review]
Please resolve conflicts.

File: rpc/common/service.proto (hunk 1 above)
Comment (ankk13) [reply rpc/common/service.proto:136]: I will change it to data

File: rpc/cache/service.proto (hunk 2 above)
Comment (ankk13) [reply rpc/cache/service.proto:69]: For new remote cache implementation, I wanted to expose getblob, to support local applylayers. But it is not needed now. Will remove it.

File: rpc/cache/service.pb.go [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
 // 	protoc-gen-go v1.27.1
-// 	protoc        v3.19.4
+// 	protoc        v3.19.1
```
Comment (knqyf263) [inline rpc/cache/service.pb.go:4]: We probably should update the version to 3.19.4.
https://github.com/aquasecurity/trivy/blob/3b0b2ed4ce481c7f058d47b0f128c906fb91a958/Dockerfile.protoc#L4

File: rpc/common/service.proto [hunk 4]
```diff
@@ -128,3 +128,10 @@ message CVSS {
   double v2_score  = 3;
   double v3_score  = 4;
 }
+
+message CustomResource {
+  string                 type     = 1;
+  string                 filepath = 2;
```
Comment (knqyf263) [inline rpc/common/service.proto:134]: Please be aligned with other fields.
https://github.com/aquasecurity/trivy/blob/d1c352461ae658aabe1c026d8c0be4ff1ded6ee7/rpc/common/service.proto#L17
https://github.com/aquasecurity/trivy/blob/d1c352461ae658aabe1c026d8c0be4ff1ded6ee7/rpc/common/service.proto#L23
https://github.com/aquasecurity/trivy/blob/d1c352461ae658aabe1c026d8c0be4ff1ded6ee7/rpc/common/service.proto#L53

File: rpc/cache/service.proto [hunk 5]
```diff
@@ -37,6 +37,7 @@ message BlobInfo {
   repeated string                  whiteout_files    = 6;
   string                           digest            = 7;
   string                           diff_id           = 8;
+  repeated common.CustomResource custom_resources    = 10;
```
Comment (knqyf263) [inline rpc/cache/service.proto:40]: The indent is broken

File: rpc/common/service.proto [hunk 6]
```diff
@@ -128,3 +128,11 @@ message CVSS {
   double v2_score  = 3;
   double v3_score  = 4;
 }
+
+message CustomResource {
+  string type      = 1;
+  string file_path = 2;
+  Layer  layer     = 3;
+
```
Comment (knqyf263) [inline rpc/common/service.proto:136]: unknown newline

**Review by knqyf263 — CHANGES_REQUESTED** [review]

File: rpc/cache/service.proto (hunk 5 above)
Comment (ankk13) [reply rpc/cache/service.proto:40]: `make fmt` is aligning as above

Comment (knqyf263) [reply rpc/cache/service.proto:40]: OK, then.

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**